apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsRegexAssertionsSpec.kt

```diff
@@ -8,41 +8,49 @@ class CharSequenceContainsRegexAssertionsSpec : Spek({
     include(StringSpec)
     include(RegexSpec)
 }) {
-    object StringSpec : ch.tutteli.atrium.specs.integration.CharSequenceContainsRegexAssertionsSpec(
+    object StringSpec : ch.tutteli.atrium.specs.integration.CharSequenceContainsRegexAssertionsSpecString(
         getNameContainsRegex(),
-        getAtLeastTriple(),
-        getAtLeastIgnoringCaseTriple(),
-        getShortcutTriple(),
-        getAtMostTriple(),
-        getAtMostIgnoringCaseTriple(),
+        getAtLeastTripleString(),
+        getAtLeastIgnoringCaseTripleString(),
+        getShortcutTripleString(),
+        getAtMostTripleString(),
+        getAtMostIgnoringCaseTripleString(),
         "◆ ", "⚬ "
     )
 
-    object RegexSpec : ch.tutteli.atrium.specs.integration.CharSequenceContainsRegexAssertionsSpec(
+    object RegexSpec : ch.tutteli.atrium.specs.integration.CharSequenceContainsRegexAssertionsSpecRegex(
         getNameContainsRegex(),
-        getAtLeastTriple(),
-        getAtLeastIgnoringCaseTriple(),
-        getShortcutTriple(),
-        getAtMostTriple(),
-        getAtMostIgnoringCaseTriple(),
+        getAtLeastTripleRegex(),
+        getAtLeastIgnoringCaseTripleRegex(),
+        getShortcutTripleRegex(),
+        getAtMostTripleRegex(),
+        getAtMostIgnoringCaseTripleRegex(),
         "◆ ", "⚬ "
     )
 
     companion object : CharSequenceContainsSpecBase() {
 
         private fun getNameContainsRegex() = "$contains with search mode $regex"
 
-        private fun getAtLeastTriple() =
+        private fun getAtLeastTripleString() =
             { what: String, times: String -> "$contains $what $atLeast $times" } to
                 ("$contains.$atLeast.$regex" to Companion::containsAtLeast)
 
+        private fun getAtLeastTripleRegex() =
+            { what: Regex, times: String -> "$contains $what $atLeast $times" } to
+                ("$contains.$atLeast.$regex" to Companion::containsAtLeast)
```

Comment:
you also need two versions of containsAtLeast (same same but different for other functions)

Reply:
Does that mean I should also overload `containsAtLeastIgnoringCase`? In so, I think I should also provide overloads for
```kotlin
@JvmName("regexIgnoringCase")
fun <T : CharSequence> CharSequenceContains.CheckerOption<T, IgnoringCaseSearchBehaviour>.regex
```
and
```kotlin
fun <T : CharSequence> CharSequenceContains.Builder<T, IgnoringCaseSearchBehaviour>.regex(
    pattern: String,
    vararg otherPatterns: String
): Expect<T>
```
in `ch/tutteli/atrium/api/fluent/en_GB/charSequenceContainsCreators.kt`.

Reply:
ah no, sorry, my fault, you don't have to overload it for those, so only two versions for containsAtLeast